com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -691,7 +674,18 @@ internal void OnDespawnObject(NetworkObject networkObject, bool destroyGameObjec
             {
                 if (NetworkManager.PrefabHandler.ContainsHandler(networkObject))
                 {
-                    NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObject);
+                    // If the custom prefab handler returns true then we destroy the parent GameObject
+                    // otherwise the custom prefab handler handles how they want to deal with despawning
+                    // the NetworkObject
+                    if(NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObject))
+                    {
+                        if (SpawnedObjects.Remove(networkObject.NetworkObjectId))
```

Comment:
I still cannot wrap my head around this. I think even if we don't destroy an object but do something else in the prefab handler we would still need to remove it from the `SpawneObjects` list. I think I'm probably just brain lagging here though.

Reply:
It gets removed under both scenarios.  The scenario where it is completely destroyed we remove it from the spawned SpawnedObjects  list(s), we completely destroy the GameObject, and then we return.  If the INetworkPrefabInstanceHandler implementation wants to keep it around, we do not destroy the GameObject and we don't return (which means around lines 696 and 792 it will remove it from the SpawnedObjects  list(s)anyway.

Both scenarios it is removed from the SpawnedObjects list(s):
```
                 if (NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObject))
                 {
                        if (SpawnedObjects.Remove(networkObject.NetworkObjectId))
                        {
                            SpawnedObjectsList.Remove(networkObject);
                        }
                        UnityEngine.Object.Destroy(gobj);
                        return;
                    }
                }
                else
                {
                    UnityEngine.Object.Destroy(gobj);
                }
            }
            // for some reason, we can get down here and SpawnedObjects for this
            //  networkId will no longer be here, even as we check this at the start
            //  of the function
            if (SpawnedObjects.Remove(networkObject.NetworkObjectId))
            {
                SpawnedObjectsList.Remove(networkObject);
            }
        }
```

Right?

Reply:
after the bug triaging meeting we had earlier today, I think this topic still worth discussing. I tend to agree with Luke in general: if we outsource spawn/despawn to a custom handler, we should not touch instances at all but just unregister them from NetworkManager/SpawnedObjects, that's it.
maybe both Luke & I can't really get the details and context here.
@NoelStephensUnity, would you like to bring us 3 together to have a chat here? we might also bring @ShadauxCat in too since she looked like she'd be interested as well.

Reply:
just an FYI here, it's going to be addressed by PR #1068

Reply:
This code no longer exists and I am marking this resolved.